File: Python/neuralnetwork1.py
"""Created by extinctsion, Edited by Tinny-Robot"""
from typing import List
import numpy as np

# initialize inputs
inputs: List[float] = [1, 2, 3, 2.5]

# initialize weights and biases
weights: List[List[float]] = [
    np.random.rand(4),
    np.random.rand(4),
    np.random.rand(4)
    ]


biases: List[float] = [0, 0, 0]

# calculate the output of each neuron using a generator
def neuron_output(_inputs: List[float], _weights: List[float], bias: float) -> float:
    return sum(n_input * weight for n_input, weight in zip(_inputs, _weights)) + bias

layer_output: List[float] = [neuron_output(inputs, neuron_weights, neuron_bias)
                              for neuron_weights, neuron_bias in zip(weights, biases)]

# print the output of the layer
print(np.round(layer_output,4))

File: C++/neuralNetwork1.cpp
#include <iostream>
#include <vector>
#include <numeric> // for std::inner_product
#include <cstdlib> // for rand()
#include <ctime>   // for seeding random

// Function to calculate neuron output
double neuron_output(const std::vector<double>& inputs, const std::vector<double>& weights, double bias) {
    return std::inner_product(inputs.begin(), inputs.end(), weights.begin(), 0.0) + bias;
}

int main() {
    // Seed the random generator
    std::srand(static_cast<unsigned>(std::time(0)));

    // Initialize inputs
    std::vector<double> inputs = {1, 2, 3, 2.5};

    // Initialize weights (randomly generated)
    std::vector<std::vector<double>> weights = {
        {static_cast<double>(std::rand()) / RAND_MAX, static_cast<double>(std::rand()) / RAND_MAX, 
         static_cast<double>(std::rand()) / RAND_MAX, static_cast<double>(std::rand()) / RAND_MAX},
        {static_cast<double>(std::rand()) / RAND_MAX, static_cast<double>(std::rand()) / RAND_MAX, 
         static_cast<double>(std::rand()) / RAND_MAX, static_cast<double>(std::rand()) / RAND_MAX},
        {static_cast<double>(std::rand()) / RAND_MAX, static_cast<double>(std::rand()) / RAND_MAX, 
         static_cast<double>(std::rand()) / RAND_MAX, static_cast<double>(std::rand()) / RAND_MAX}
    };

    // Initialize biases
    std::vector<double> biases = {0, 0, 0};

    // Calculate the output of each neuron
    std::vector<double> layer_output;
    for (size_t i = 0; i < weights.size(); ++i) {
        layer_output.push_back(neuron_output(inputs, weights[i], biases[i]));
    }

    // Print the output of the layer
    std::cout << "Layer output: ";
    for (double output : layer_output) {
        std::cout << std::round(output * 10000) / 10000 << " "; // Round to 4 decimal places
    }
    std::cout << std::endl;

    return 0;
}
